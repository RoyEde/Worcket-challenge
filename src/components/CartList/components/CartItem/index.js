import React from 'react';
import PropTypes from 'prop-types';

import Button from '../../../Button';

const CartItem = ({ item, onDelete }) => {
  return (
    <li className="cartlist__item">
      <h4 className="cartlist__item__descriptor">{item}</h4>
      <Button aria-label={`Delete item ${item}`} danger onClick={onDelete}>Delete</Button>
    </li>
  );
};

CartItem.propTypes = {
  
};

export default CartItem;