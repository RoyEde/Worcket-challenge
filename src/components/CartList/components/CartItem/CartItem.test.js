import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import CartItem from './index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CartItem item="test" onDelete={() => {}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshots match', () => {
  const component = renderer.create(<CartItem item="test" onDelete={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  tree.props.item = 'test 2';
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
