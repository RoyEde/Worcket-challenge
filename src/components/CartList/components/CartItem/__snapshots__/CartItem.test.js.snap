// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match 1`] = `
<li
  className="cartlist__item"
>
  <h4
    className="cartlist__item__descriptor"
  >
    test
  </h4>
  <button
    aria-label="Delete item test"
    className="button   button--danger"
    onClick={[Function]}
  >
    Remove
  </button>
</li>
`;

exports[`snapshots match 2`] = `
<li
  className="cartlist__item"
>
  <h4
    className="cartlist__item__descriptor"
  >
    test
  </h4>
  <button
    aria-label="Delete item test"
    className="button   button--danger"
    onClick={[Function]}
  >
    Remove
  </button>
</li>
`;
